src/ci/install.sh

```diff
@@ -58,6 +58,35 @@ case "${TEST_TYPE}" in
           echo 'JAVA_OPTS="$JAVA_OPTS -Dcassandra.available_processors=2"' >> ~/.ccm/test/node$i/conf/cassandra-env.sh
         done
         ;;
+    "sidecar")
+        ccm create test -v $CASSANDRA_VERSION > /dev/null
+        # use "2:0" to ensure the first datacenter name is "dc1" instead of "datacenter1", so to be compatible with CircleCI tests
+        ccm populate --vnodes -n 2:0 > /dev/null
+        for i in `seq 1 2` ; do
+          sed -i 's/jmxremote.authenticate=true/jmxremote.authenticate=false/' ~/.ccm/test/node$i/conf/cassandra-env.sh
+          sed -i 's/#MAX_HEAP_SIZE="4G"/MAX_HEAP_SIZE="256m"/' ~/.ccm/test/node$i/conf/cassandra-env.sh
+          sed -i 's/#HEAP_NEWSIZE="800M"/HEAP_NEWSIZE="200M"/' ~/.ccm/test/node$i/conf/cassandra-env.sh
+          sed -i 's/_timeout_in_ms:.*/_timeout_in_ms: 60000/' ~/.ccm/test/node$i/conf/cassandra.yaml
+          sed -i 's/start_rpc: true/start_rpc: false/' ~/.ccm/test/node$i/conf/cassandra.yaml
+          sed -i 's/cross_node_timeout: false/cross_node_timeout: true/' ~/.ccm/test/node$i/conf/cassandra.yaml
+          sed -i 's/concurrent_reads: 32/concurrent_reads: 4/' ~/.ccm/test/node$i/conf/cassandra.yaml
+          sed -i 's/concurrent_writes: 32/concurrent_writes: 4/' ~/.ccm/test/node$i/conf/cassandra.yaml
+          sed -i 's/concurrent_counter_writes: 32/concurrent_counter_writes: 4/' ~/.ccm/test/node$i/conf/cassandra.yaml
+          sed -i 's/num_tokens: 256/num_tokens: 4/' ~/.ccm/test/node$i/conf/cassandra.yaml
+          sed -i 's/auto_snapshot: true/auto_snapshot: false/' ~/.ccm/test/node$i/conf/cassandra.yaml
+          sed -i 's/enable_materialized_views: true/enable_materialized_views: false/' ~/.ccm/test/node$i/conf/cassandra.yaml
+          sed -i 's/internode_compression: dc/internode_compression: none/' ~/.ccm/test/node$i/conf/cassandra.yaml
+          sed -i 's/# file_cache_size_in_mb: 512/file_cache_size_in_mb: 1/' ~/.ccm/test/node$i/conf/cassandra.yaml
+          echo 'phi_convict_threshold: 16' >> ~/.ccm/test/node$i/conf/cassandra.yaml
+          if  echo "$CASSANDRA_VERSION" | grep -q "trunk"  ; then
+            sed -i 's/start_rpc: true//' ~/.ccm/test/node$i/conf/cassandra.yaml
+            echo '-Dcassandra.max_local_pause_in_ms=15000' >> ~/.ccm/test/node$i/conf/jvm-server.options
+            sed -i 's/#-Dcassandra.available_processors=number_of_processors/-Dcassandra.available_processors=2/' ~/.ccm/test/node$i/conf/jvm-server.options
+          else
+            sed -i 's/start_rpc: true/start_rpc: false/' ~/.ccm/test/node$i/conf/cassandra.yaml
+          fi
+        done
+        ;;
```

Comment:
what's different about this to `"ccm"` ?
(it seems to be a fair bit to duplicate just for the switch case in `script.sh` ?)

Reply:
it is removing all the parts around setting up JMX authentication, since it's not activated in local JMX mode.

Reply:
agree to put lines 67-87 into a separate function that both switch cases can use.

Reply:
Done